usage/app/controllers/UsageApi.scala

```diff
@@ -321,19 +322,40 @@ class UsageApi(
       "requestId" -> RequestLoggingFilter.getRequestId(req),
       "image-id" -> mediaId,
     )
-    usageTable.queryByImageId(mediaId).map(usages => {
-      usages.foreach(usageTable.deleteRecord)
-    }).recover {
-      case error: BadInputException =>
-        logger.warn(logMarker, "UsageApi returned an error.", error)
-        respondError(BadRequest, "image-usage-delete-failed", error.getMessage)
-      case error: Exception =>
-        logger.error(logMarker, "UsageApi returned an error.", error)
-        respondError(InternalServerError, "image-usage-delete-failed", error.getMessage)
-    }
 
-    val updateMessage = UpdateMessage(subject = DeleteUsages, id = Some(mediaId))
-    notifications.publish(updateMessage)
+    var photoSalesUsage = false
+    var filteredUsages = ListBuffer[MediaUsage]()
+
+    usageTable.queryByImageId(mediaId)
```

Comment:
We should be able to write this in a functional way, avoiding the need for `var` and mutable data structures. Something like this... ?

```
    usageTable.queryByImageId(mediaId)
      .map(usages => {
        val hasPhotoSalesUsage = (usage: MediaUsage) => usage.syndicationUsageMetadata.exists(_.partnerName == "Capture")
        val usagesWithoutPhotoSales = usages.filter(!hasPhotoSalesUsage(_))

        usagesWithoutPhotoSales.foreach(usage => usageTable.deleteRecord(usage))

        if (usages.exists(hasPhotoSalesUsage)) {
          usagesWithoutPhotoSales foreach { usage =>
            val usageId = Some(usage.grouping + "_" + usage.usageId.toString)
            val updateMessage = UpdateMessage(subject = DeleteSingleUsage, id = Some(mediaId), usageId = usageId)
            notifications.publish(updateMessage)
          }
        } else {
          val updateMessage = UpdateMessage(subject = DeleteUsages, id = Some(mediaId))
          notifications.publish(updateMessage)
        }
      })
```

I'm not sure if it makes the logic flow of the code a little easier to follow